tests/testthat/test-reporter.R

```diff
@@ -77,4 +82,111 @@ test_that("reporters produce consistent output", {
   save_report("silent")
   save_report("rstudio")
   save_report("junit", reporter = createJunitReporterMock())
+
+  # Test that MultiReporter can write to two different places
+  tap_file <- tempfile()
+  save_report("summary", reporter = MultiReporter$new(list(
+    SummaryReporter$new(show_praise = FALSE, omit_dots = FALSE),
+    TapReporter$new(file=tap_file))
+  ))
+  expect_identical(read_lines(tap_file),
+    read_lines(test_path("reporters", "tap.txt")))
+})
+
+
+test_report_to_file <- function(name, reporter = find_reporter_one(name, ...),
+                                output_file, error_regexp=NA, ...) {
+  # output_file is where we expect output to be written to, whether we pass it
+  # as an argument to Reporter$new() (here, via the ...), or whether it is set
+  # in an option.
+  path <- test_path("reporters", paste0(name, ".txt"))
+  expect_silent(
+    expect_error(
+      test_reporter(reporter),
+      error_regexp
+    )
+  )
+  expect_identical(read_lines(output_file), read_lines(path))
+}
+
+test_that("reporters accept a 'file' arugment and write to that location", {
+  output <- tempfile()
+  test_report_to_file("check", file = output, error_regexp=NULL, output_file = output)
```

Comment:
You only need to test each reporter once, i.e. not multiple times for `SummaryReporter`